src/lib/utilities/Popup/popup.ts

```diff
@@ -182,6 +183,7 @@ export function popup(node: HTMLElement, args: PopupSettings) {
 		if (autoUpdateCleanup) autoUpdateCleanup();
 	}
 	function closeOnFocusOut(e: FocusEvent) {
+		console.log('called');
```

Comment:
I realize you're still working on this so I thought I'd call this out while I saw it. Careful about leaving in console calls ;)

Reply:
whoops! Thanks for pointing that out!

Reply:
that being said, I think everything is done now , I just realised some of my testing around non-interactive elements with tabindexes vs interactive elements was wrong. They're still a little different, in that clicking on a non interactive element with tabindex=0 will open the popup, but a button wouldn't. I think a note in the docs would be better than checking the element type of the node, but I'm open to a better idea